lib/PrintAsClang/DeclAndTypePrinter.cpp

```diff
@@ -490,10 +493,31 @@ class DeclAndTypePrinter::Implementation
 
   void visitEnumDeclCxx(EnumDecl *ED) {
     assert(owningPrinter.outputLang == OutputLanguageMode::Cxx);
+    ClangSyntaxPrinter syntaxPrinter(ED->getASTContext(), os);
+
+    if (owningPrinter.isEmptyEnum(ED)) {
+      syntaxPrinter.printParentNamespaceForNestedTypes(
+          ED, [&](raw_ostream &os) {
+            syntaxPrinter.printNamespace(
+                cxx_translation::getNameForCxx(ED),
+                [ED, this](llvm::raw_ostream &) {
```

Comment:
why does this work despite not using the ostream?

Reply:
Here, we are calling member functions where the parent class already has the same `ostream`. Unfortunately, it is not really regular when we pass ostreams as parameters versus store it in an object.